src/nix/modules/service/nixos-init.nix

```diff
@@ -37,9 +37,15 @@ in
       "/sys/fs/cgroup:/sys/fs/cgroup:ro"
     ];
     service.tmpfs = [
-      "/run"          # noexec is fine because exes should be symlinked from elsewhere anyway
-      "/run/wrappers" # noexec breaks this intentionally
-    ] ++ lib.optional (config.nixos.evaluatedConfig.boot.tmpOnTmpfs) "/tmp:exec,mode=777";
+      let
+        useTmpfs = # suppress option rename trace message, it was done in 23.05 release
+          if (config.nixos.evaluatedConfig.boot.tmp != null) then config.nixos.evaluatedConfig.boot.tmp.useTmpfs
```

Comment:
Would this really be `null`?

Reply:
According to https://github.com/NixOS/nixpkgs/pull/204534, it should be.

Reply:
I've tested it a bit, this change will work as expected, in `evaluatedConfig`, `boot.tmp` will be attrset only if this option was renamed.